bears/elm/ElmLintBear.py

```diff
@@ -0,0 +1,37 @@
+from shutil import which
+
+from coalib.bearlib.abstractions.Linter import linter
+from dependency_management.requirements.NpmRequirement import NpmRequirement
+
+
+@linter(executable='elm-format',
+        use_stdin=True,
```

Comment:
Firstly, IO related to a normal source file will be statistically insignificant compared to the IO of starting `elm-format`.  And "language in language" can use temporary named files (and will need to, as many linters dont support stdin), so there is no problems there.

It isnt easy for the external command to handle stdin.  It is typically done as an afterthought, and not done properly.
If there is a problem reading a file, fallback mechanism can be attempted in normal exception handling of the linter, and it is fairly normal to do this.
If there is a problem reading stdin, ... the program must have kept a literal bytestream in order to re-attempt to read the file.  Most programs dont bother, letting their stdin support lag behind their filename handling.
Then there are all the problems with stdin encoding detection, which is not the same as file handling because the language runtimes believe they should 'help' the three default handles, especially wrt utf8.

For a very practical example of the types of problems we'll be encountering, I offer https://gitlab.com/pycqa/flake8/issues?scope=all&search=stdin&state=all
And that is just Python.)
Every other runtime has its own lovely oddities.
(I expect I could find at least five difficult bugs in coala's subprocess creation, in 24 hrs)

Also filenames help with logic around paths, which obviously cant occur with stdin. (some of the above flake8 bugs are because of this; flake8 allows passing the filename to be used in conjunction with stdin to work around some of these problems).

There is also the aspect of `overwrite_file` below needing the filename anyway, so why not always do that instead of having two code paths that achieve the same thing.